### DISCUSSION TITLE
Organize aliases by category/tag

### DISCUSSION THREAD

**Discussion by nguyenkims** [simple-login/app#398] [Feature Requests]
Moved from https://trello.com/c/aHQAjFUQ/50-organize-aliases-by-category-tag

**Comment by gitmaker032** [comment]
I'm looking forward to the implementation of this feature. When there are a lot of aliases (a separate alias for each service) you want to systematize them. For example, I have many aliases for social networks. I would like to click on the tag and see all the aliases associated with this category. 

Below I have visualized what this might look like:

![simplelogin-tags](https://user-images.githubusercontent.com/67278783/114884569-63180900-9df5-11eb-9cf2-73c3d1b0a0bc.png)
[View "Before and After"](https://i.imgur.com/EfK3Zcq.png)

I made a few other changes that I think might be useful:
1. Added a tags field (when entering tags, a popup appears that displays similar tags and allows you to add a new tag).
2. Added display of tags near the alias activity switch (click on a tag to turn on the filter for this tag, colors are set randomly when creating the tag).
3. Removed save buttons (automatic saving or one button instead).
4. Removed the "Add alias to favorites" switch (the link button instead).
5. Added "Show less" switch after opening the details of the alias.
6. All the fields are aligned in the same style.

**Reply by nguyenkims** [reply]
Thanks for the design idea! Do you have a suggestion on the UI to add a tag to an alias?

**Reply by gitmaker032** [reply]
@nguyenkims I've updated my previous comment, there I showed what it might look like. I also described the changes made and added a before and after comparison :)

**Reply by nguyenkims** [reply]
The design looks great! We'll probably proceed in 2 steps: first adding the tagging feature and after, improving the UI (auto save, fields alignment, tag coloring) to avoid potential regression :).

**Comment by ghost** [comment]
Is there an update on the creation of alias groups?

**Reply by nguyenkims** [reply]
@reis-i-cumhur we don't have any update yet (the team is on holiday in August).

**Reply by MeasurementRick** [reply]
> 
> 
> @reis-i-cumhur we don't have any update yet (the team is on holiday in August).

Hi. Just thought I'd check in on this feature request and see if there are any updates on progress / timing? I upgraded from free to Premium recently, so between adding aliases for a custom domain that I own + starting to use more SL aliases in general for my accounts I'm seeing the challenges of viewing all of my SL aliases. It would be great to have folders and/or tags (ideally both) for organization and for filtering.

**Reply by nguyenkims** [reply]
In the meantime you can use the advanced search feature or filter by a custom domain. It isn't as complete as a tag system though.

**Comment by LenKiMo** [comment]
I think it can not only adding label, but also making Alias page showing catalogue by default. So that when open Alias page in dashboard, it would more simple and clean.

**Reply by Vnzq4Qje3zzU** [reply]
Definetly. Tags could also replace the ready-to-input empty note fields in the web dashboard (only display the notes if there is something there and hide it inside "see more" when empty), and you could add some small button "+ add tag", making everything more compact and clean.

**Comment by ITechGeek81** [comment]
Any status update?

**Reply by nguyenkims** [reply]
This feature is part of this year roadmap. The first step to make it happen to convert the dashboard into a SPA first which will take some time.

**Reply by MeasurementRick** [reply]
That's great to hear. Thanks for the update!

**Reply by gammexane** [reply]
Great to hear, this is the only feature missing to make sl perfect to me.

**Comment by Coder84619** [comment]
Any update on the progress?

**Reply by nguyenkims** [reply]
@Coder84619 we don't have any update on this feature yet. We'll finalize our roadmap for 2023 Q1 soon and hope we can make this feature into the roadmap.

**Comment by Coder84619** [comment]
@nguyenkims I think a related feature would be auto-tagging of aliases, by filtering on a user-defined alias sub-string. Let me explain. For finance accounts lets say I want to use the string 'bg85' (e.g chase.bg85@mydomain.com, wellsfargo.bg85@mydomain.com), then in the SimpleLogin UI when I define a tag I could optionally set a filter string of bg85. Then if I'm using the catch-all feature, I can make up finance accounts on the fly with the string 'bg85' and they will get auto-tagged on your back end. I also use these alias strings in email client filtering, using the same logic. Any inbound emails with 'bg85' in the recipient address get routed to a Finance inbox via a rule in Outlook. 

And taking that one step further, in your browser plugin or alternative keyboard the UI should allow me to select/search for a tag and when it auto-generates the alias, it would insert the associated string automatically (if one was defined).

**Comment by DennisDyallo** [comment]
Hi, just thought Id share how Im using SimpleLogin.
I prefix all my aliases with a category tag. For instance: newsletters-glassnode@mydomain.simplelogin.com, services-proton@mydomain.simplelogin.com, shopping-amazon@mydomain.simplelogin.com. It'd be nice if SimpleLogin could somehow categorize them and let me route all emails matching that prefix to a specified inbox (thanks to inbox filters in Proton)

**Comment by roughwood** [comment]
Any updates on this feature? I have way too many aliases and it's becoming really cumbersome to manage them

**Comment by rehanone** [comment]
This is a much needed feature as the growing number of aliases quickly get out of hand. Any update on this would be a great news.

**Comment by galure** [comment]
I like this idea, but this seems to take a long time to develop. It's been 2 years!
Could we just have folders at least? Simply being able to group aliases would make life easier.
I use subdomains to categorize aliases. Even something as trivial as "sorting by subdomains" would be welcomed.

**Comment by gammexane** [comment]
Hello, a year has passed since my last update. I already have more than 300 alias and my dashboard is a mess. We really need a way to sort them. Alias, folders, whatever... but please!

**Comment by HyperCriSiS** [comment]
Has the development stopped?
The most wanted features are years old... :(

**Comment by ghost** [comment]
Hi there,

Does anyone know if there is any update on this feature request?

Thanks

**Comment by nguyenkims** [comment]
This is still in progress. The reason it takes so long as is it requires changing SL from a "classic" web app to a single page application (SPA). 

This is actually a quite big project as SL wasn't designed to become SPA. However this SPA rewrite is already in progress and we hope to be able to finish it this year. It'll also allow to normalize the way mobile & web clients interact with SL server.

**Reply by HyperCriSiS** [reply]
When this is finished, will there be generally more updates? Because it seems everything is on hold for many years now, no real progress.

**Reply by DennisDyallo** [reply]
Must it be a SPA then? A lot of users are using SL through Proton. An API to categorize aliases and redirect routing could be utilized within the Proton mail suite. :)

**Comment by radutomy** [comment]
Any updates on this? :)

**Comment by eu6en** [comment]
Hi, is there any ETA on this feature? 
Can we have it, pretty please? 🥺

**Comment by DennisDyallo** [comment]
Must it be a SPA then? A lot of users are using SL through Proton. An API to categorize aliases and redirect routing could be utilized within the Proton mail suite. :)

**Reply by DennisDyallo** [reply]
Shouldnt take 3 years to convert any app to a single page application..??

**Comment by Rocket-Quack** [comment]
Is there an update on the implementation. Or also of possible folder structures to sort aliases better? By now it has been almost 4 years, which rather gives the impression that the feature has been put on hold

**Comment by DennisDyallo** [comment]
Wondering the same thing. No updates..

**Comment by usrnm317** [comment]
I would love to see this too. I've never used tags (would prefer folders, I think) but could figure it out.

**Comment by hiccups1980** [comment]
We really need this please. 🙏

**Comment by grndng** [comment]
I have the feeling that it won't do much but I thought I'd chip in and say that I'd also love to have this feature. It would make organizing things much easier in the interface. Also I've noticed that this is in "Discussion" and not an Issue. Is it redundant to create an issue for this?

**Comment by rhertlein** [comment]
Please add some way to categorize or organize aliases into folders or something.  Pinning is nice, but being able to organize my growing list so I'm not overwhelmed with a huge list when I only want to see aliases used for shopping or online accounts would be nice.  I'd prefer to drill down to the alias I want rather than be presented with everything all at once from the start.

**Reply by kollette27** [reply]
I haven’t done it this way, but I’m finding the need to do so already and I’ve only been a pro member for a month. However, my theory is that you can stylize your “on the fly” aliases like “shopping.amazon@domain” or “social.facebook@domain”, “subscriptions.netflix@domain”. Then, use filters to quickly find those you’re looking for. But the @domain assumes you’re using your own domain, although it could work with the SL random generator I guess. I haven’t really used it long enough to understand ALL its functions, but so far it’s been quite useful.

**Comment by DennisDyallo** [comment]
I felt like creating this issue on the subject. Not sure if maintainers are reading these forums any longer:

[[Feature Request] Alias Organization (Tags/Folders)](https://github.com/simple-login/app/issues/2657)

**Comment by mielas-mp3** [comment]
Hope to see this soon! Or at least a response on this would be appreciated!
